openedx_learning/apps/authoring/backup_restore/zipper.py

```diff
@@ -75,41 +86,51 @@ def create_zip(self, path: str) -> None:
-                        folders_already_created.add(component_version_folder)
-
-                    for entity_version in entity.component.versions.all():
-                        component_number_version_folder = component_version_folder / f"v{entity_version.version_num}"
-                        # Create a folder for each version of the component. Example: "v1", "v2", etc.
-                        if component_number_version_folder not in folders_already_created:
-                            self.create_folder(component_number_version_folder, zipf)
-                            folders_already_created.add(component_number_version_folder)
-
-                        # Add the static folder inside the component version folder
-                        static_folder = component_number_version_folder / "static"
-                        if static_folder not in folders_already_created:
-                            self.create_folder(static_folder, zipf)
-                            folders_already_created.add(static_folder)
+                    self.create_folder(component_version_folder, zipf)
+
+                    # ------ COMPONENT VERSIONING -------------
+                    # Focusing on draft and published versions
+
+                    # Get the draft and published versions
+                    current_draft: Optional[Draft] = getattr(entity, "draft", None)
+                    current_published: Optional[Published] = getattr(entity, "published", None)
+
+                    draft_version: Optional[PublishableEntityVersion] = getattr(current_draft, "version", None)
+                    published_version: Optional[PublishableEntityVersion] = getattr(current_published, "version", None)
+
+                    # Creating draft version folder
+                    if draft_version:
+                        # Create a folder for the draft version
+                        draft_version_number = f"v{draft_version.version_num}"
+                        draft_version_folder = component_version_folder / draft_version_number
+                        self.create_folder(draft_version_folder, zipf)
+
+                        # Add static folder for the draft version
+                        static_folder = draft_version_folder / "static"
+                        self.create_folder(static_folder, zipf)
+
+                    if published_version and published_version != draft_version:
+                        # Create a folder for the published version
+                        published_version_number = f"v{published_version.version_num}"
+                        published_version_folder = component_version_folder / published_version_number
+                        self.create_folder(published_version_folder, zipf)
+
+                        # Add static folder for the published version
+                        static_folder = published_version_folder / "static"
+                        self.create_folder(static_folder, zipf)
```

Comment:
Nit: This is redundant with the first block. So you could do something like:

```python
versions_to_write = [draft_version] if draft_version else []
if published_version and published_version != draft_version:
    versions_to_write.append(published_version)

for version in versions to write:
    # the other code
```

Just to reduce the odds of these blocks accidentally drifting somehow.